src/protocol/encryption_info.h

```diff
@@ -53,8 +55,8 @@ class EncryptionInfo {
   bool                decrypt_valid() const             { return m_decryptValid; }
 
   void                set_obfuscated()                  { m_obfuscated = true; m_encrypted = m_decryptValid = false; }
-  void                set_encrypt(const RC4& encrypt)   { m_encrypt = encrypt; m_encrypted = m_obfuscated = true; }
-  void                set_decrypt(const RC4& decrypt)   { m_decrypt = decrypt; m_decryptValid = true; }
+  void                set_encrypt(RC4 encrypt)          { m_encrypt = std::move(encrypt); m_encrypted = m_obfuscated = true; }
+  void                set_decrypt(RC4 decrypt)          { m_decrypt = std::move(decrypt); m_decryptValid = true; }
```

Comment:
Seems these should be `set_*(RC4&& *crypt)`.

Reply:
hmm? they're still moved.

Reply:
By using rvalue reference you enforce the use of `std::move`.

Reply:
sure. although there's no advantage here. As I mentioned earlier, it's hard to remove copies without pointers.

reading https://clang.llvm.org/extra/clang-tidy/checks/modernize/pass-by-value.html , before this PR, the RC4 call uses the copy constructor because of const ref. When it takes a value parameter, the compiler is free to choose copy or move. Although the way I modified it, it's move.

Although reading OpenSSL code is convincing me custom copy or move constructors need to be written...

Reply:
My point was that using `RC&&` makes it so there's less chance that that any future changes to the code misses the need requirement that RC4 is never copied.

It's about using how you write the code to express what your intentions are, and here we want to say RC4 is never copied.

That's why imo it should be using rvalue reference, not value, even if the latter results in the correct output.

Reply:
Done.